### ISSUE TITLE
Create helper functions

### ISSUE THREAD

**Issue by PaulWoitaschek** [square/kotlinpoet#1072]
I use several helper functions on the CodeBlock.Builder.
Would you accept a PR on that?

My currently heavy used functions are:
```kotlin
import com.squareup.kotlinpoet.CodeBlock

internal fun CodeBlock.Builder.withIndent(action: CodeBlock.Builder.() -> Unit): CodeBlock.Builder {
  indent()
  action()
  unindent()
  return this
}

internal fun CodeBlock.Builder.withControlFlow(
  controlFlow: String,
  vararg args: Any?,
  action: CodeBlock.Builder.() -> Unit
): CodeBlock.Builder {
  beginControlFlow(controlFlow, *args)
  action()
  endControlFlow()
  return this
}
```

This makes writing code way more pleasant because it's no longer unnecessary to remember where a block started and where it ended.

An example looks like this:

```kotlin
      buildCodeBlock {
        withControlFlow("return when(id)") {
          stories.forEach { story ->
            add(
              buildCodeBlock {
                withControlFlow("%T.${story.storyId} ->", storyIdClassName) {
                  addStatement("%T(", regularStoryClassName)
                  withIndent {
                    addStatement("proOnly = %L,", story.proOnly)
                    addStatement("backendKey = %S,", story.backendKey)
                    addStatement("color = %T.%L,", colorClassName, story.color.capitalize(Locale.ROOT))
                    addStatement("pages = listOf(")
                    withIndent {
                      [...]
                    }
                }
          }
        }
      }
    
```

**Comment by JakeWharton** [comment]
The second one does not support more complex conditionals that use `else if` or `else` so I am opposed to that one. One key principle of the Android ktx extensions was that you shouldn't have to switch completely away from the extension as your needs grew.

First one seems fine. I have tons of these to make the library more like a DSL, too.

Whether we want to be in the business of curating extensions is another question. In practice, everyone has things like this and everyone thinks theirs should be part of the library (once you start shipping them).

**Comment by Egorand** [comment]
Closed by #1073
